### PR TITLE
[REEF-1438] Validate Exception in spun off System.Threading.Tasks.Task does not trigger FailedEvaluator event

### PR DESCRIPTION
**Pull request by tcNickolas**
JIRA:
  [REEF-1438](https://issues.apache.org/jira/browse/REEF-1438)

Pull request:
  This closes #

### CODE REVIEW

**Comment by tcNickolas** [comment]
This change also cleans up several tests which I've looked at while adding a new one. 

A good AppVeyor build (for at least one job) can be found at https://ci.appveyor.com/project/tcNickolas/reef/build/484-REEF-1438

**Comment by markusweimer** [comment]
I left a question over on JIRA.

File: lang/cs/Org.Apache.REEF.Tests/Functional/Failure/User/UnhandledTaskExceptionInTaskTest.cs [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+                    }
+
+                    if (inner.Message != ExpectedExceptionMessage)
+                    {
+                        throw new Exception("Incorrect Exception message, got message: " + value.AsError().Message);
+                    }
+
+                    value.GetActiveContext().Value.Dispose();
+
+                    if (_numFailedTasksReceived == 2)
```
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.Tests/Functional/Failure/User/UnhandledTaskExceptionInTaskTest.cs]: Put the log before context dispose to make sure it is logged before driver is shut down.

**Comment by jwang98052** [comment]
I completed a pass.

**Comment by tcNickolas** [comment]
I've addressed the comment and updated the PR. Apache AppVeyor seems to be stuck again, a good build is at my branch: https://ci.appveyor.com/project/tcNickolas/reef/build/486-REEF-1438

**Comment by jwang98052** [comment]
LGTM. I will merge it.

**Pull request closed without merging**